yt/visualization/volume_rendering/utils.py

```diff
@@ -74,6 +74,36 @@ def new_volume_render_sampler(camera, render_source):
     sampler = VolumeRenderSampler(*args, **kwargs)
     return sampler
 
+def new_light_render_sampler(camera, render_source):
+    params = ensure_code_unit_params(camera._get_sampler_params(render_source))
+    params.update(transfer_function=render_source.transfer_function)
+    params.update(transfer_function=render_source.transfer_function)
+    params.update(num_samples=render_source.num_samples)
+    args = (
+        np.atleast_3d(params['vp_pos']),
+        np.atleast_3d(params['vp_dir']),
+        params['center'],
+        params['bounds'],
+        params['image'],
+        params['x_vec'],
+        params['y_vec'],
+        params['width'],
+        params['transfer_function'],
+        params['num_samples'],
+    )
+    kwargs = {'lens_type': params['lens_type']}
+    if "camera_data" in params:
+        kwargs['camera_data'] = params['camera_data']
+    if render_source.zbuffer is not None:
+        kwargs['zbuffer'] = render_source.zbuffer.z
+        args[4][:] = np.reshape(render_source.zbuffer.rgba[:], \
+            (camera.resolution[0], camera.resolution[1], 4))
+    else:
+        kwargs['zbuffer'] = np.ones(params['image'].shape[:2], "float64")
+
+    sampler = LightSourceRenderSampler(*args, **kwargs, light_dir=[-1, 1, 1])
```

Comment:
Right now the tests are failing because this isn't allowed on Python2. Instead you'll need to insert `light_dir` into `kwargs`.

Also I think it should be possible for users to set `light_dir` so we should probably think about how to expose that to users.

Reply:
It occurs to me that another way to do this would be to add a `light_direction` property to the `VolumeSource` that defaults to `None` and can be set to a 3-element array-like. If it's set you call the new sampler, if it's not set you call the regular volume rendering sampler.

If we did this I guess we might also need to add a light source to the mesh renderer and I don't know how straightforward that would be or if it even makes any sense. I guess you could also raise an error if it's set and you're rendering an unstructured mesh.

Reply:
I've made a local change I'll commit here in a bit, but what I ended up doing was setting the light_dir to the normal vector of the camera. This actually works out really well for most situations, so when I make the suggested changes and make it more user exposed, this will be what it defaults to instead of an arbitrary number. I should have some time over next weekend to work on this.